### PR TITLE
Documentation Fixes and Additions

### PR DESCRIPTION
**Pull request by ubombar**
This patch contains some adding and fixes in the documentation part. I am looking forward for your feedback.

### CODE REVIEW

File: docs/README.md [hunk 1]
```diff
@@ -1,8 +1,40 @@
-# EdgeNet documentation
+# EdgeNet Documentation
 
-This is where we develop documentation for EdgeNet, the Kubernetes-based edge cloud software that
-powers the [EdgeNet testbed](http://www.edge-net.org/), providing dozens of nodes at the internet's
-edge to distributed systems and network measurement researchers worldwide.
+## What is EdgeNet?
+EdgeNet, the open-source globally distributed edge cloud for Internet researchers, is based on industry-standard Cloud software, using [Docker](https://www.docker.com/) for containerization and [Kubernetes](https://kubernetes.io/) for deployment and node management. The source code of EdgeNet can be found [here](https://github.com/EdgeNet-project/edgenet).
```
Comment (bsenel) [inline docs/README.md]: Can you please rewrite this part?

Comment (bsenel) [reply docs/README.md]: Here, as this doc is on GitHub, we may introduce the source code first, then mention that we provide researchers a testbed for the not-for-profit research.

File: docs/README.md [hunk 2]
```diff
@@ -1,8 +1,40 @@
-# EdgeNet documentation
+# EdgeNet Documentation
 
-This is where we develop documentation for EdgeNet, the Kubernetes-based edge cloud software that
-powers the [EdgeNet testbed](http://www.edge-net.org/), providing dozens of nodes at the internet's
-edge to distributed systems and network measurement researchers worldwide.
+## What is EdgeNet?
+EdgeNet, the open-source globally distributed edge cloud for Internet researchers, is based on industry-standard Cloud software, using [Docker](https://www.docker.com/) for containerization and [Kubernetes](https://kubernetes.io/) for deployment and node management. The source code of EdgeNet can be found [here](https://github.com/EdgeNet-project/edgenet). 
+
+Additionally, we provide a testbed for all researchers. To open an account please refer to the [Registering a tenant into EdgeNet](tenant_registration.md). For getting more info about the project visit the official [EdgeNet-Project](https://www.edge-net.org/) website.
```
Comment (bsenel) [inline docs/README.md]: I would not mention opening an account here as you introduce components below. It is better to directly take those who want to register a tenant with the EdgeNet testbed to the testbed website.

File: docs/README.md [hunk 3]
```diff
@@ -1,8 +1,40 @@
...
+* [Node Contribution](custom_resources.md#node-contribution)
+* [Role Request](custom_resources.md#role-request)
+* [Selective Deployment](custom_resources.md#selective-deployment)
+* [Slice](custom_resources.md#slice)
+* [Slice Claim](custom_resources.md#slice-claim)
+* [Subnamespace](custom_resources.md#subnamespace)
+* [Tenant](custom_resources.md#tenant)
+* [Tenant Request](custom_resources.md#tenant-request)
+* [Tenant Resource Quota](custom_resources.md#tenant-resource-quota)
+* [VPN Peer](custom_resources.md#vpn-peer)
```
Comment (bsenel) [inline docs/README.md]: Can we introduce categories here?

Like below:
1) Multitenancy
- Tenant
- Tenant Resource Quota
- Subnamespace
- Slice
- Slice Claim
- Tenant Request
- Role Request
- Cluster Role Request

2) Multi-provider
- Node Contribution
- VPN Peer

3) Location-based node selection
- Selective Deployment

File: docs/README.md (hunk 1 above)
Comment (bsenel) [reply docs/README.md]: We should separate these two here clearly; the EdgeNet source code and the EdgeNet testbed. Then the rest of doc should be only about the source code.

File: docs/cluster_creation.md [hunk 4]
```diff
@@ -0,0 +1,41 @@
+# Cluster creation from EdgeNet source code
```
Comment (bsenel) [inline docs/cluster_creation.md]: I would express it like **Deploying EdgeNet on a Kubernetes cluster**.

File: docs/cluster_creation.md [hunk 5]
```diff
@@ -0,0 +1,41 @@
...
+
+This documentation describes how to create a working EdgeNet cluster in your environment. You can refer to this tutorial for creating a sandbox, local, or production cluster for your desires.
+
+## Technologies you will use
+You will need a Kubernetes cluster to extend it to support EdgeNet. There are many ways to setup a Kubernetes cluster all of which require different knowledge. You can refer [here](https://kubernetes.io/docs/setup/production-environment/tools/kubeadm/create-cluster-kubeadm/) for how to set up a Kubernetes cluster with `kubeadm`. Also, to set up a local cluster you can use [minikube](https://minikube.sigs.k8s.io/docs/) or [k3s](https://docs.k3s.io/installation) (be aware EdgeNet is not yet tested with k3s).
```
Comment (bsenel) [inline docs/cluster_creation.md]: This should be in another section that is about how to create a cluster.

File: docs/cluster_creation.md [hunk 6]
```diff
@@ -0,0 +1,41 @@
...
+
+If you want to compile EdgeNet from the source you need to install [`kompose`](https://kompose.io/#:~:text=Kompose%20is%20a%20conversion%20tool,as%20Kubernetes%20(or%20OpenShift).) command-line tool.
+
+## What will you do
+
+Before the tutorial, it is assumed you have an access to a working Kubernetes cluster with a valid kubeconfig and `kubectl` command-line tool installed.
```
Comment (bsenel) [inline docs/cluster_creation.md]: This should go into the section of **Technologies you will use**.

File: docs/cluster_creation.md [hunk 7]
```diff
@@ -0,0 +1,41 @@
...
+
+There are two stages for installing EdgeNet to a Kubernetes cluster. First, it is necessary to install the [CRDs](https://kubernetes.io/docs/concepts/extend-kubernetes/api-extension/custom-resources/) to the cluster. This operation will create the objects which are required for EdgeNet to function. Then, the custom resources will be deployed. This allows CRDs to be created deleted and altered, and thus EdgeNet to work.
```
Comment (bsenel) [inline docs/cluster_creation.md]: Please rewrite this part.

File: docs/cluster_creation.md [hunk 8]
```diff
@@ -0,0 +1,41 @@
...
+
+```
+    kubectl -f apply ./build/yamls/kubernetes/all-in-one.yaml
+```
+
+This command creates all of the objects in Kubernetes including the deployments. Thus, it takes some time for them to start working.
+
+<!-- *TODO: all-in-one.yaml analysis* -->
+
+<!-- *TODO: How to compile EdgeNet from source?* -->
```
Comment (bsenel) [inline docs/cluster_creation.md]: Can you please provide subsections for each [deployment file](https://github.com/EdgeNet-project/edgenet/tree/main/build/yamls/kubernetes)?

File: docs/custom_resources.md [hunk 9]
```diff
@@ -0,0 +1,39 @@
...
+## Tenant Request
+Tenant request is sent to any user with the aim of joining the cluster for research purposes. The request should contain the contact information of the tenant as well as the address of the establishment or research institution. 
+
+To create a tenant request in EdgeNet refer to the [tutorial](tenant_registration.md).
+
+## Tenant Resource Quota
+In EdgeNet each tenant owns a set of resources. This means tenants shares resources in a multi-tenant cluster. each tenant resource quota describes a set of resources to be added and removed from the access of the tenant.
+
+## VPN Peer
+EdgeNet is a distributed edge cluster that has nodes all around the world. To connect these nodes in the same network a VPN is used. VPN peer objects contain information about different nodes IP addresses, endpoints, and public keys.
```
Comment (bsenel) [inline docs/custom_resources.md]: Can you please put them in categories, as I mentioned in one of the previous comments?

Comment (bsenel) [reply docs/custom_resources.md]: We need to revisit the descriptions too.

**Review by bsenel — CHANGES_REQUESTED** [review]
Thank you for your effort! It seems in good shape, but it requires more work to be ready.

Can you please address my comments?

File: docs/custom_resources.md (hunk 9 above)
Comment (ubombar) [reply docs/custom_resources.md]: Hello, I have rearranged the content on this page and rephrased some of the descriptions. How is it now?

File: docs/cluster_creation.md (hunk 8 above)
Comment (ubombar) [reply docs/cluster_creation.md]: I have written the section for one of the yamls. If its adequate I can do it for the others as well.

**Comment by ubombar** [comment]
As I mentioned in the #180 I will also add the fedmanctl documentation, tutorial, and installation guide to this patch.

**Comment by ubombar** [comment]
I have also added the fedmanctl documentation with this PR.

**Comment by ubombar** [comment]
I added the latest updates on the `fedmanctl` to the documentation.

**Pull request merged**